Source/Core/Core/Debugger/CodePatch.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Core/Debugger/CodePatch.h"
+
+#include "Core/PowerPC/PowerPC.h"
+
+CodePatchBase::~CodePatchBase()
+{
+}
+
+CodePatch::CodePatch(u32 address, const std::vector<u8>& patch) : m_address(address), m_patch(patch)
+{
+}
+
+bool CodePatch::Apply()
+{
+  if (GetState() != State::SOURCE)
+    return false;
+
+  for (u32 index = 0, size = static_cast<u32>(m_patch.size()); index < size; index += 1)
+  {
+    m_source.push_back(PowerPC::HostRead_U8(m_address + index));
+    PowerPC::HostWrite_U8(m_patch[index], m_address + index);
+  }
+
+  return true;
+}
+
+bool CodePatch::Remove()
+{
+  if (GetState() != State::PATCH)
+    return false;
+
+  for (u32 index = 0, size = static_cast<u32>(m_source.size()); index < size; index += 1)
+    PowerPC::HostWrite_U8(m_source[index], m_address + index);
+  m_source.clear();
+
+  return true;
+}
+
+CodePatch::State CodePatch::GetState()
+{
+  if (m_source.empty())
+    return State::SOURCE;
+
+  for (u32 index = 0, size = static_cast<u32>(m_patch.size()); index < size; index += 1)
```

Comment:
...and down here.

I'd personally also like some braces around the `for` since it nests deeper than one level inside; but I don't insist on it if others consider it ok.